### DISCUSSION TITLE
Dynamically fill MUI-table

### DISCUSSION THREAD

**Discussion by alinawinter** [neuefische/questions#579] [Web] [answered]
Someone please help me, because I am going crazy.

# Situation
I have included a Material-UI table into my code, which I want to dynamically fill with all different kinds of data that I have collected and calculated (some I still have to calculate). The table renders and looks beautiful :-D

# Problem
What is tricky, is dynamically filling the table with data, which arises from the fact that the part of the table that receives the data (the „rows“-variable) is in the components-file („WeeklyScoreTable.js“)  but not part of the „WeeklyScoreTable“-component or any other function. That makes it difficult to pass data as props. 
I managed to render some data („foodCategories“) into it, that I had directly imported from my database („db.js“), that I didn’t have to pass as props. The other data that I want to include, however, is trapped in other components and I cannot pass it as prop, because the „rows“-variable doesn’t receive props or I don’t know how to do it. Hence, I have not managed to get it into the variable outside of my component, in my components file. The data is called „sumOfActualWeeklyConsumptionByFoodCategories“  and „recommendedConsumptionByFoodCategoryBasedOnNumberQuizzes“

# Errormessage
„sumOfActualWeeklyConsumptionByFoodCategories“ (the data that I want to include) is not defined“ (because I don’t manage to get it into the file and use it)

# What I tried so far:
I have tried different things like including the rows-variable into my „WeeklyScoreTable“-component, using useEffect or making a function out of the rows-variable, always ending up with error messages and the realization that it is not possible that way.

I created a PR of my current state: https://github.com/alinawinter/capstone-project/pull/23/files
This is the deployment, you can only see the table, when you do one quiz: https://capstone-project-git-add-weekly-detail-results-alinawinter.vercel.app/
The issue is in component/Score/WeeklyScoreTable.js. The data that I want to import is currently at pages/index.js.

I really hope I've just become blind, looking for a solution and there is maybe even an easy fix for it. Thank you already very much for your help.

**Comment by F-Kirchhoff** [comment]
You don't accept your props in the WeeklyScoreTable: https://github.com/alinawinter/capstone-project/pull/23/files#diff-95e71e0d55f30f4aeb15073d24a116555a16a10fe4368c87cc98de2d21bb1461
You pass them into the component in your main page, but the Component has only an emtpy Object `{}` as props. here you need to accept the props by using the names you used in the main page.

**Reply by alinawinter** [reply]
Hey thank's for checking! I think you misunderstood my question or I'm really confused by now. The issue that I am having is not getting the props inside the WeeklyScoreTable component but into the "rows"-Variable, which is outside of the WeeklyScoreTable-component and according to what I tried, needs to stay out of it.

**Comment by alinawinter** [comment, marked as answer]
I solved it with the help of @F-Kirchhoff, thank you :-) 
We integrated the "rows"variable and connected function into the component and changed the variable-declaration so that the connected function could be deleted and didn't cause any issues for the component anymore. This way it became possible to receive the props and make my external data visible in the table. My key take away is not to neglect to fully understand the logic of MUI-tables and not be afraid to change them if necessary! Thanks alot, Felix!! :-)
